crates/router/src/compatibility/stripe/payment_intents/types.rs

```diff
@@ -127,31 +134,96 @@ impl From<StripePaymentMethodDetails> for payments::PaymentMethodData {
 
 #[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
 pub struct Shipping {
-    pub address: Option<payments::AddressDetails>,
-    pub name: Option<String>,
+    pub address: AddressDetails,
+    pub name: Option<Secret<String>>,
     pub carrier: Option<String>,
-    pub phone: Option<pii::Secret<String>>,
-    pub tracking_number: Option<pii::Secret<String>>,
+    pub phone: Option<Secret<String>>,
+    pub tracking_number: Option<Secret<String>>,
+}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct AddressDetails {
+    pub city: Option<String>,
+    pub country: Option<api_enums::CountryAlpha2>,
+    pub line1: Option<Secret<String>>,
+    pub line2: Option<Secret<String>>,
+    pub postal_code: Option<Secret<String>>,
+    pub state: Option<Secret<String>>,
 }
 
 impl From<Shipping> for payments::Address {
     fn from(details: Shipping) -> Self {
         Self {
             phone: Some(payments::PhoneDetails {
                 number: details.phone,
-                country_code: details.address.as_ref().and_then(|address| {
-                    address.country.as_ref().map(|country| country.to_string())
-                }),
+                country_code: details.address.country.map(|country| country.to_string()),
             }),
-            address: details.address,
+            address: Some(payments::AddressDetails {
+                city: details.address.city,
+                country: details.address.country,
+                line1: details.address.line1,
+                line2: details.address.line2,
+                zip: details.address.postal_code,
+                state: details.address.state,
+                first_name: details.name,
+                line3: None,
+                last_name: None,
+            }),
+        }
+    }
+}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct MandateData {
+    pub customer_acceptance: CustomerAcceptance,
+}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct CustomerAcceptance {
+    #[serde(rename = "type")]
+    pub acceptance_type: AcceptanceType,
+    pub accepted_at: Option<PrimitiveDateTime>,
+    pub online: Option<OnlineMandate>,
+}
+
+#[derive(Default, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum AcceptanceType {
+    Online,
+    #[default]
+    Offline,
+}
+
+#[derive(Default, Eq, PartialEq, Debug, serde::Deserialize, serde::Serialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct OnlineMandate {
+    pub ip_address: Secret<String, IpAddress>,
+    pub user_agent: String,
+}
+
+impl From<MandateData> for payments::MandateData {
```

Comment:
why are we using a new trait for mandate data, we already have a ForeignTryFrom trait for mandates for this in line 684.